packages/gatsby-source-contentful/src/__tests__/gatsby-node.js

```diff
@@ -0,0 +1,346 @@
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
+  })
+
+  it(`should add a new blogpost and update linkedNodes`, async () => {
+    const newBlogPostId = `a1b2c2345def___c5Q1RtFHGRDbvbY5rZbaaZP`
+    const newBlogPostIdNl = `a1b2c2345def___c5Q1RtFHGRDbvbY5rZbaaZP___nl`
+    const blogPostEntry =
+      startersBlogFixture.createBlogPost.currentSyncData.entries[0]
+
+    fetch
+      .mockReturnValueOnce(startersBlogFixture.initialSync)
+      .mockReturnValueOnce(startersBlogFixture.createBlogPost)
+
+    // initial sync
+    await gatsbyNode.sourceNodes({
+      actions,
+      store,
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
+
+    expect(getNode(newBlogPostId)).toBeUndefined()
+    expect(getNode(newBlogPostIdNl)).toBeUndefined()
+
+    // add new blog post
+    await gatsbyNode.sourceNodes({
+      actions,
+      store,
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
+
+    const newBlogPost = getNode(newBlogPostId)
+    const newBlogPostNl = getNode(newBlogPostIdNl)
+    // const author = getNode(newBlogPost[`author___NODE`])
+    // const authorNl = getNode(newBlogPostNl[`author___NODE`])
+
+    const expectedObjectUS = {
+      contentful_id: blogPostEntry.sys.id.substr(1),
+      title: blogPostEntry.fields.title[`en-US`],
+      slug: blogPostEntry.fields.slug[`en-US`],
+      node_locale: `en-US`,
+      description___NODE: createNodeId(`${newBlogPostId}descriptionTextNode`),
+      body___NODE: createNodeId(`${newBlogPostId}bodyTextNode`),
+      heroImage___NODE: normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.heroImage[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `en-US`,
+      }),
+      author___NODE: normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.author[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `en-US`,
+      }),
+    }
+    const expectedObjectNl = {
+      ...expectedObjectUS,
+      node_locale: `nl`,
+      description___NODE: createNodeId(`${newBlogPostIdNl}descriptionTextNode`),
+      body___NODE: createNodeId(`${newBlogPostIdNl}bodyTextNode`),
+      heroImage___NODE: normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.heroImage[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `nl`,
+      }),
+      author___NODE: normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.author[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `nl`,
+      }),
+    }
+
+    expect(newBlogPost).toMatchObject(expectedObjectUS)
+    expect(newBlogPostNl).toMatchObject(expectedObjectNl)
+
+    // test author links
+    // TODO fix
```

Comment:
I assume this is what you mentioned the other day about problem in current state of contentful plugin - could there be some context around the problem here - maybe linked github issue?